### PR TITLE
add (init) --dev documentation

### LINKED ISSUE #3781: dapr init -k --dev command needs more documentation
To make setting up a development environment on Kubernetes easier, the [dapr init -k command](https://v1-12.docs.dapr.io/reference/cli/dapr-init/) now supports a --dev option which deploy Redis and Zipkin containers to Kubernetes identical to the self-hosted mode. 

- This Initialize Dapr in your local environment overview should be updated to make it clear that this option is now available for K8s. https://v1-12.docs.dapr.io/getting-started/install-dapr-selfhost/. Either create a new topic or maybe enhance this one to add Kubernetes
- Example needed for dapr init -k --dev in the examples section
Add an example here  https://v1-12.docs.dapr.io/reference/cli/dapr-init/#kubernetes-environment

### PR DESCRIPTION
**Pull request by mikeee**
This made sense to link it to the existing kubernetes init documentation and add a subsection with the `--dev` flag documented.

**Please follow this checklist before submitting:**
- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://docs.dapr.io/contributing/contributing-overview/#developer-certificate-of-origin-signing-your-work))
- [x] [Read the contribution guide](https://docs.dapr.io/contributing/docs-contrib/contributing-docs/)
- [x] Commands include options for Linux, MacOS, and Windows within codetabs
- [x] New file and folder names are globally unique
- [x] Page references use shortcodes instead of markdown or URL links
- [x] Images use HTML style and have alternative text
- [x] Places where multiple code/command options are given have codetabs

In addition, please fill out the following to help reviewers understand this pull request:

## Description

1. Implements `dapr init -k --dev` documentation in kubernetes operations
2. Add a `--dev` example in the init cli reference.

## Issue reference

Please reference the issue this PR will close: #3781

### CODE REVIEW

File: daprdocs/content/en/operations/hosting/kubernetes/kubernetes-deploy.md [hunk 1]
```diff
@@ -72,6 +72,51 @@ The `-k` flag initializes Dapr on the Kubernetes cluster in your current context
     dapr dashboard -k -n <your-namespace>
     ```
 
+
+ #### Install Dapr from the offical Dapr Helm chart (with development flag)
+
+Adding the `--dev` flag initializes Dapr on the Kubernetes cluster on your current context with the addition of Redis and Zipkin deployments.
+
+The steps are similar to the above, save for the inclusion of the `--dev` flag being appended to the init command like so:
```
Comment (msfussell) [inline daprdocs/content/en/operations/hosting/kubernetes/kubernetes-deploy.md]:
```suggestion
The steps are similar to the above, save for the inclusion of the `--dev` flag being appended to the init command:
```

File: daprdocs/content/en/operations/hosting/kubernetes/kubernetes-deploy.md [hunk 2]
```diff
@@ -72,6 +72,51 @@ The `-k` flag initializes Dapr on the Kubernetes cluster in your current context
...
+✅  Deploying the Dapr dashboard with latest version to your cluster...
+✅  Deploying the Dapr Redis with latest version to your cluster...
+✅  Deploying the Dapr Zipkin with latest version to your cluster...
+ℹ️  Applying "statestore" component to Kubernetes "default" namespace.
+ℹ️  Applying "pubsub" component to Kubernetes "default" namespace.
+ℹ️  Applying "appconfig" zipkin configuration to Kubernetes "default" namespace.
+✅  Success! Dapr has been installed to namespace dapr-system. To verify, run `dapr status -k' in your terminal. To get started, go here: https://aka.ms/dapr-getting-started
+ ```
+
+After a short period of time (or using the `--wait` flag and specifying an amount of time to wait) you should be able to check that the Redis and Zipkin components have been stood up correctly.
```
Comment (msfussell) [inline daprdocs/content/en/operations/hosting/kubernetes/kubernetes-deploy.md]:
```suggestion
After a short period of time (or using the `--wait` flag and specifying an amount of time to wait) you can check that the Redis and Zipkin components have been deployed to the cluster.
```

File: daprdocs/content/en/operations/hosting/kubernetes/kubernetes-deploy.md [hunk 3]
```diff
@@ -72,6 +72,51 @@ The `-k` flag initializes Dapr on the Kubernetes cluster in your current context
...
+
+Adding the `--dev` flag initializes Dapr on the Kubernetes cluster on your current context with the addition of Redis and Zipkin deployments.
+
+The steps are similar to the above, save for the inclusion of the `--dev` flag being appended to the init command like so:
+
+ ```bash
+ dapr init -k --dev
+ ```
+
+The expected output will be:
```
Comment (msfussell) [inline daprdocs/content/en/operations/hosting/kubernetes/kubernetes-deploy.md]:
```suggestion
Expected output:
```

File: daprdocs/content/en/operations/hosting/kubernetes/kubernetes-deploy.md [hunk 4]
```diff
@@ -72,6 +72,51 @@ The `-k` flag initializes Dapr on the Kubernetes cluster in your current context
...
+✅  Success! Dapr has been installed to namespace dapr-system. To verify, run `dapr status -k' in your terminal. To get started, go here: https://aka.ms/dapr-getting-started
+ ```
+
+After a short period of time (or using the `--wait` flag and specifying an amount of time to wait) you should be able to check that the Redis and Zipkin components have been stood up correctly.
+
+```bash
+kubectl get pods --namespace default
+```
+
+Expected Output:
```
Comment (msfussell) [inline daprdocs/content/en/operations/hosting/kubernetes/kubernetes-deploy.md]:
```suggestion
Expected output:
```

File: daprdocs/content/en/getting-started/install-dapr-selfhost.md [hunk 5]
```diff
@@ -10,6 +10,8 @@ aliases:
 
 Now that you've [installed the Dapr CLI]({{<ref install-dapr-cli.md>}}), use the CLI to initialize Dapr on your local machine.
 
+To initialize dapr in your local (or remote) kubernetes cluster. [How to initialize dapr with redis and zipkin]({{<ref "kubernetes-deploy.md#install-dapr-from-the-official-dapr-helm-chart-with-development-flag">}})
```
Comment (msfussell) [inline daprdocs/content/en/getting-started/install-dapr-selfhost.md]:
```suggestion
To initialize Dapr in your local (or remote) kubernetes cluster for development including the Redis and Zipkin containers listed above see [how to initialize Dapr for development on Kubernetes]({{<ref "kubernetes-deploy.md#install-dapr-from-the-official-dapr-helm-chart-with-development-flag">}})
```

Comment (msfussell) [reply daprdocs/content/en/getting-started/install-dapr-selfhost.md]: I suggest that this paragraph above ^^ be move to after this line in the doc

"4. Running a Dapr placement service container instance for local actor support."

That way you have some understand of what will happen first with Zipkin and Redis.

**Review by msfussell — COMMENTED** [review]
Looking good!Here are a few comments.

File: daprdocs/content/en/getting-started/install-dapr-selfhost.md (hunk 5 above)
Comment (mikeee) [reply daprdocs/content/en/getting-started/install-dapr-selfhost.md]: @msfussell I'm thinking that maybe keeping it where it is but putting it into an alert with the kubernetes title may be a better fit as this diverges from the rest of the guide and should provide a helpful hint towards the kubernetes init process

File: daprdocs/content/en/getting-started/install-dapr-selfhost.md [hunk 6]
```diff
@@ -10,6 +10,10 @@ aliases:
 
 Now that you've [installed the Dapr CLI]({{<ref install-dapr-cli.md>}}), use the CLI to initialize Dapr on your local machine.
 
+{{% alert title="Kubernetes Development Environment" color="primary" %}}
+To initialize Dapr in your local (or remote) kubernetes cluster for development including the Redis and Zipkin containers listed above see [how to initialize Dapr for development on Kubernetes]({{<ref "kubernetes-deploy.md#install-dapr-from-the-official-dapr-helm-chart-with-development-flag">}})
```
Comment (hhunter-ms) [inline daprdocs/content/en/getting-started/install-dapr-selfhost.md]:
```suggestion
To initialize Dapr in your local or remote Kubernetes cluster for development (including the Redis and Zipkin containers listed above), see [how to initialize Dapr for development on Kubernetes]({{<ref "kubernetes-deploy.md#install-dapr-from-the-official-dapr-helm-chart-with-development-flag">}})
```

File: daprdocs/content/en/operations/hosting/kubernetes/kubernetes-deploy.md [hunk 7]
```diff
@@ -72,6 +72,51 @@ The `-k` flag initializes Dapr on the Kubernetes cluster in your current context
     dapr dashboard -k -n <your-namespace>
     ```
 
+
+ #### Install Dapr from the offical Dapr Helm chart (with development flag)
+
+Adding the `--dev` flag initializes Dapr on the Kubernetes cluster on your current context with the addition of Redis and Zipkin deployments.
```
Comment (hhunter-ms) [inline daprdocs/content/en/operations/hosting/kubernetes/kubernetes-deploy.md]:
```suggestion
Adding the `--dev` flag initializes Dapr on the Kubernetes cluster on your current context, with the addition of Redis and Zipkin deployments.
```

File: daprdocs/content/en/operations/hosting/kubernetes/kubernetes-deploy.md [hunk 8]
```diff
@@ -72,6 +72,51 @@ The `-k` flag initializes Dapr on the Kubernetes cluster in your current context
...
+
+The steps are similar to the above, save for the inclusion of the `--dev` flag being appended to the init command:
```
Comment (hhunter-ms) [inline daprdocs/content/en/operations/hosting/kubernetes/kubernetes-deploy.md]:
```suggestion
The steps are similar to [installing from the Dapr Helm chart](#install-dapr-from-an-official-dapr-helm-chart), except for appending the `--dev` flag to the `init` command:
```

File: daprdocs/content/en/operations/hosting/kubernetes/kubernetes-deploy.md [hunk 9]
```diff
@@ -72,6 +72,51 @@ The `-k` flag initializes Dapr on the Kubernetes cluster in your current context
...
+✅  Deploying the Dapr dashboard with latest version to your cluster...
+✅  Deploying the Dapr Redis with latest version to your cluster...
+✅  Deploying the Dapr Zipkin with latest version to your cluster...
+ℹ️  Applying "statestore" component to Kubernetes "default" namespace.
+ℹ️  Applying "pubsub" component to Kubernetes "default" namespace.
+ℹ️  Applying "appconfig" zipkin configuration to Kubernetes "default" namespace.
+✅  Success! Dapr has been installed to namespace dapr-system. To verify, run `dapr status -k' in your terminal. To get started, go here: https://aka.ms/dapr-getting-started
+ ```
+
+After a short period of time (or using the `--wait` flag and specifying an amount of time to wait) you can check that the Redis and Zipkin components have been deployed to the cluster.
```
Comment (hhunter-ms) [inline daprdocs/content/en/operations/hosting/kubernetes/kubernetes-deploy.md]:
```suggestion
After a short period of time (or using the `--wait` flag and specifying an amount of time to wait), you can check that the Redis and Zipkin components have been deployed to the cluster.
```

File: daprdocs/content/en/reference/cli/dapr-init.md [hunk 10]
```diff
@@ -162,6 +162,12 @@ dapr uninstall --all --network mynet
 dapr init -k
 ```
 
+Using the `--dev` flag will initialize dapr in dev mode which includes Zipkin and Redis.
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/cli/dapr-init.md]:
```suggestion
Using the `--dev` flag initializes Dapr in dev mode, which includes Zipkin and Redis.
```

**Review by hhunter-ms — CHANGES_REQUESTED** [review]
quick review

**Review by hhunter-ms — APPROVED** [review]

**Review by msfussell — APPROVED** [review]
LGTM

File: daprdocs/content/en/getting-started/install-dapr-selfhost.md (hunk 5 above)
Comment (msfussell) [reply daprdocs/content/en/getting-started/install-dapr-selfhost.md]: @mikeee  -given the text where is says deploys zipkin and Redis, it does fit better in an alert further down the page. I move it there, in an edit, PTAL and tell me. IMO this is good to merge now

**Comment by mikeee** [comment]
> LGTM

@msfussell The repositioning looks good to me 👍

**Pull request merged**